build: Beautify cmake files

mikew-lunarg opened the pull request.

karl-lunarg commented:
I'm not a big fan of the format program converting a single line comment to a multi-line inline comment.

```cmake
# Fix up the entire widget so that the bracket can do what it wants
set(x 1)
```
new:
```cmake
set(x 1)           # Fix up the entire widget
                       # so that the bracket can
                       # do what it wants.
```

Is there anyway to turn this off?

I'm positive enough about the rest of this to put up with this problem.

mikew-lunarg commented:
Determined that is a bug in cmake-format, have submitted [an issue](https://github.com/cheshirekow/cmake_format/issues/50).

There are a couple workaround options, let's talk.

mikew-lunarg commented:
@karl-lunarg PTAL
Addressed the bad comment reflow via a workaround, added docs to CONTRIBUTING.md, and refined the formatting with some embedded markup.
